### PR TITLE
feat: display dates in local format

### PR DESCRIPTION
**Pull request by rofe**
Display dates on cards and in articles matching the date format of the corresponding language.

### CODE REVIEW

File: scripts.js [hunk 1]
```diff
@@ -192,10 +192,26 @@ window.blog = function() {
...
+    dateLocale = 'en-US'; // default to US date format
+  }
+  if (dateLocale === LANG.BR) {
+    dateLocale = 'pt-BR';
+  }
+  if (dateLocale === LANG.JP) {
+    dateLocale = 'ja-JP';
+  }
+  const pageName = window.location.pathname.split('/').pop().split('.')[0];
+  if (pageName === 'uk' || pageName === 'apac') {
```
Comment (kptdobe) [inline scripts.js:207]: We should use the sessionStorage info which is computed for the region picker. Like this, we can show the uk date format on articles too.

See https://github.com/adobe/theblog/blob/master/scripts/common.js#L740

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: scripts.js (hunk 1 above)
Comment (rofe) [reply scripts.js:207]: TBH, I tried that at first. But unfortunately, the current logic is incorrect: it doesn't set the region iwhen the user selects a different region in the dropdown, as I would have expected, but only during retrieval of the selected region. This means it's currently always one step behind and can't be used to influence display logic:
1. go to blog.adobe.com and select UK in the region dropdown
2. the UK homepage loads, but `blog-selected-language` is still undefined (or whatever it was before)
3. then select German in the region dropdown
4. the German homepage loads, but `blog-selected-language` is now `en_uk`

I was thinking about digging into this, but then I realized that there is a host of other problems when using the sessionStorage:
- We would need another mapping anyway in order to make it work (`Date.toLocalDateString()` doesn't understand `en_uk` or `en_apac`, but expects `en-UK`).
- And even if it worked, that locale might still be too sticky: imagine if a visitors navigates to a different language manually after having used the region dropdown, just to check what's on the other side, and then goes back using the browser's back button: we currently don't set the correct regional language on every homepage, so from that point in time, the user would see all content with dates formatted in the language they once selected in the region dropdown, and they can't get rid of it anymore.

Bottom line: we would have to refactor a lot of the region picker if we were to use its selected locale. That's why I decided to go with a simple implementation for the date formatting for now.

**Review by kptdobe — APPROVED** [review]
